WordPress/gutenberg
# Interface: Use plugin context hook instead of HoC

## What?
Similar to #66350.

PR updates components in the `interface` package to use the `usePluginContext` hook instead of HOC.

## Why?
Using the context hook should be preferred over HOC.

## Testing Instructions

This indirectly tests the affected components, but this is the only case when plugin context is available.

1. Open a post.
2. Run the test snippet.
3. Confirm that the new sidebar is available.
4. Confirm that the icon is inherited from the plugin registration.

<details><summary>Snippet</summary>
<p>

```js
var el = React.createElement;
var Fragment = wp.element.Fragment;
var PluginSidebar = wp.editor.PluginSidebar;
var PluginSidebarMoreMenuItem = wp.editor.PluginSidebarMoreMenuItem;
var registerPlugin = wp.plugins.registerPlugin;

function Component() {
	return el(
		Fragment,
		{},
		el(
			PluginSidebarMoreMenuItem,
			{
				target: 'test-sidebar-name',
			},
			'Test Sidebar'
		),
		el(
			PluginSidebar,
			{
				name: 'test-sidebar-name',
				title: 'Test Sidebar',
			},
			'Content of the sidebar'
		)
	);
}
registerPlugin( 'test-plugin-sidebar', {
	icon: 'carrot',
	render: Component,
} );
```

</p>
</details> 

### Testing Instructions for Keyboard
Same.

## Screenshots or screencast 
![CleanShot 2024-10-23 at 14 30 58](https://github.com/user-attachments/assets/d260d119-b582-434c-8d7a-507ff650c28a)
